src/host/inputBuffer.cpp

```diff
@@ -218,6 +220,28 @@ void InputBuffer::FlushAllButKeys()
     _storage.erase(newEnd, _storage.end());
 }
 
+void InputBuffer::SetTerminalConnection(_In_ ITerminalOutputConnection* const pTtyConnection)
+{
+    this->_pTtyConnection = pTtyConnection;
+}
+
+void InputBuffer::PassThroughWin32MouseRequest(bool enable)
+{
+    if (_pTtyConnection)
+    {
+        if (enable)
+        {
+            LOG_IF_FAILED(_pTtyConnection->WriteTerminalW(L"\x1b[?1003h"));
+            LOG_IF_FAILED(_pTtyConnection->WriteTerminalW(L"\x1b[?1006h"));

```

Comment:
You could probably put these in one request! Since we support DECSET with multiple parameters...

```suggestion
            LOG_IF_FAILED(_pTtyConnection->WriteTerminalW(L"\x1b[?1003;1006h"));
```

(I checked xterm- it looks like it supports multiple DECSET/DECRST modes in one mode line? I can't corroborate that with the VT510 manual. @j4james likely knows if this is standards-compliant... he explicitly made it work in #7799!)

Reply:
Yeah, this is standards compliant. It's documented in the DEC STD-070 manual, section 2.4.5.1 Set Mode.
> The SM control provides a means of placing one or more of the terminal mode values into the set state... The parameter list sent with a single instance of this control should not exceed the maximum allowable for the parser implementation. 

They recommend support for up to 16 parameters, but some terminals will have a lower limit than that, and I think there are also some that don't support multiple mode parameters at all. Unfortunately I don't have access to my terminal collection at the moment, so I'm not sure how big a problem that is.

Reply:
Excellent, thank you

Reply:
Thank you! We only write once now